desc/compute/_basis_vectors.py

```diff
@@ -9,686 +20,3414 @@
+    dim=3,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["R", "R_r", "R_rt", "R_t", "Z_rt", "omega_r", "omega_rt", "omega_t"],
+)
+def _e_sub_rho_t(params, transforms, profiles, data, **kwargs):
+    data["e_rho_t"] = jnp.array(
+        [
+            -data["R"] * data["omega_t"] * data["omega_r"] + data["R_rt"],
+            data["omega_t"] * data["R_r"]
+            + data["R_t"] * data["omega_r"]
+            + data["R"] * data["omega_rt"],
+            data["Z_rt"],
+        ]
+    ).T
+    return data
+
+
+@register_compute_fun(
+    name="e_rho_z",
+    label="\\partial_{\\zeta} \\mathbf{e}_{\\rho}",
+    units="m",
+    units_long="meters",
+    description="Covariant Radial basis vector, derivative wrt toroidal coordinate",
+    dim=3,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["R", "R_r", "R_rz", "R_z", "Z_rz", "omega_r", "omega_rz", "omega_z"],
+)
+def _e_sub_rho_z(params, transforms, profiles, data, **kwargs):
+    data["e_rho_z"] = jnp.array(
+        [
+            -data["R"] * (1 + data["omega_z"]) * data["omega_r"] + data["R_rz"],
+            (1 + data["omega_z"]) * data["R_r"]
+            + data["R_z"] * data["omega_r"]
+            + data["R"] * data["omega_rz"],
+            data["Z_rz"],
+        ]
+    ).T
+    return data
+
+
+@register_compute_fun(
+    name="e_rho_rr",
+    label="\\partial_{\\rho}{\\rho} \\mathbf{e}_{\\rho}",
+    units="m",
+    units_long="meters",
+    description=(
+        "Covariant Radial basis vector, second derivative wrt radial and radial"
+        " coordinates"
+    ),
+    dim=3,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["R", "R_r", "R_rr", "R_rrr", "Z_rrr", "omega_r", "omega_rr", "omega_rrr"],
+)
+def _e_sub_rho_rr(params, transforms, profiles, data, **kwargs):
+    data["e_rho_rr"] = jnp.array(
+        [
+            -3 * data["R_r"] * data["omega_r"] ** 2
+            - 3 * data["R"] * data["omega_r"] * data["omega_rr"]
+            + data["R_rrr"],
+            3 * (data["omega_r"] * data["R_rr"] + data["R_r"] * data["omega_rr"])
+            + data["R"] * (-data["omega_r"] ** 3 + data["omega_rrr"]),
+            data["Z_rrr"],
+        ]
+    ).T
+    return data
+
+
+@register_compute_fun(
+    name="e_rho_rrr",
+    label="\\partial_{\\rho}{\\rho}{\\rho} \\mathbf{e}_{\\rho}",
+    units="m",
+    units_long="meters",
+    description=(
+        "Covariant Radial basis vector, third derivative wrt radial coordinate"
+    ),
+    dim=3,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=[
+        "R",
+        "R_r",
+        "R_rr",
+        "R_rrr",
+        "R_rrrr",
+        "Z_rrrr",
+        "omega_r",
+        "omega_rr",
+        "omega_rrr",
+        "omega_rrrr",
+    ],
+)
+def _e_sub_rho_rrr(params, transforms, profiles, data, **kwargs):
```

Comment:
I added the third derivatives below the second derivatives because it's helpful to see the previous derivative to check for correctness. If git is struggling with the diff this like this maybe I should move it to the bottom then we can tell GitHub to compare with `rc/toroidal_angle`.